### PR TITLE
Prepare v4 docs adding an empty migration guide

### PR DESCRIPTION
**Pull request by gsteel**
|    Q          |   A
|-------------- | ------
| Documentation | yes

### Description

Duplicates existing docs and empties the migration guide for v3 -> v4. Updates mkdocs with new navigation

### CODE REVIEW

File: docs/book/v4/migration.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Migration Guide
+
+## From v3 to v4
```
Comment (froschdesign) [inline docs/book/v4/migration.md]:
```suggestion
# Migration from Version 3 to 4
```

File: mkdocs.yml [hunk 2]
```diff
@@ -5,16 +5,20 @@ nav:
     - v3:
       - "Console Helper": v3/console-helper.md
       - Migration: v3/migration.md
+    - v4:
+        - "Console Helper": v4/console-helper.md
+        - Migration: v4/migration.md
```
Comment (froschdesign) [inline mkdocs.yml]: An minor version may also need descriptions, therefore:

```suggestion
        - Migration:
           - "Migration from Version 3 to 4": v4/migration/v3-to-v4.md
```

- Reference: https://github.com/laminas/documentation/issues/13
- Example: https://docs.laminas.dev/laminas-form/v3/migration/v2-to-v3/

File: docs/book/v4/console-helper.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+## When to upgrade
+
+`ConsoleHelper` is deliberately simple, and assumes that your primary need for
+console tooling is for output considerations.
+
+If you need to parse complex argument strings, we recommend using
+[laminas-console](https://docs.laminas.dev/laminas-console/)/[zf-console](https://github.com/zfcampus/zf-console)
+or [symfony/console](http://symfony.com/doc/current/components/console.html),
+as these packages provide those capabilities, as well as far more colorization
+and console feature detection facilities.
```
Comment (froschdesign) [inline docs/book/v4/console-helper.md]: Should be reduce to symfony/console.

**Comment by gsteel** [comment]
Thanks for the help @froschdesign - and the link to the component docs outline. Does this look good to merge now?

**Review by froschdesign — APPROVED** [review]

**Pull request merged**